### ISSUE TITLE
[BUG]: Get Details of SyncIQ Policy fails with:  Invalid value for `snapshot_sync_pattern`, length must be greater than or equal to `1`"

### ISSUE THREAD

**Issue by davisglenn** [dell/ansible-powerscale#14]
**Describe the bug**
All I am trying to do is get the Details of a SyncIQ policy.  I followed the Example in the Repo.
I get an error saying:
**_"msg": "Get details of SyncIQ policy AutomationRecovery failed with error : Invalid value for `snapshot_sync_pattern`, length must be greater than or equal to `1`"_**

**To Reproduce**
1. Step 1 - create the playbook with the values

```
  - name: Get SyncIQ policy details
     dellemc.powerscale.dellemc_powerscale_synciqpolicy:
      onefs_host: "{{ dr_onefs_host }}"
      api_user: "{{ dr_api_user }}"
      api_password: "{{ dr_api_password }}"
      verify_ssl: "{{ verify_ssl }}"
      policy_name: "AutomationRecovery"
      snapshot_sync_pattern: "*file*"
      state: "present"
    register: result
```

2. Step 2 - Run the playbook
3. Step 3 - See error

**Expected behavior**
I expect the Playbook not to fail  and the details to be available in the register variable.

**System Information (please complete the following information):**
 - OS/Version: RHEL 8.3
 - Ansible Version: 2.9.6
 - Python Version 3.6.8  

**Additional Info**
I tried this without the snapshot_sync_pattern as well since this is only doing a GET Details there should be no other information needed other than the Policy Name.
Also, NONE of the other modules work in the Examples shown in the Repo.

Lastly, the REST API Calls all work.
It would be great if the Ansible modules could use the REST API calls instead of the Python Modules.  In this case you would not need to install the Python modules.

Thanks for your help!
glenn.

**Comment by dcsk1234** [comment]
Hi Glenn,
We have tried to Fetch synciq policy created with different patterns as values for snapshot_sync_pattern parameter and in all the cases, GET call is working and we are not able to hit the issue

However when we tried to create synciq policy with blank value for snapshot_sync_pattern using playbooks, then error message mentioned in issue is seen and we are not able to create synciq policy. Also tried from OneFS UI, creating a policy with blank pattern but it defaults value to '*' and when we tried fetching the same policy details using playbook, we are getting the details

In a nutshell we are able to see the error reported only during create scenario. Could you please share the parameters with which you have created synciq policy so that we will try the same in our environment?

[Get_details_synciq_policy.txt](https://github.com/dell/ansible-powerscale/files/7924837/Get_details_synciq_policy.txt)

**Comment by davisglenn** [comment]
How about we do this:  Here is my CREATE task which fails like every other task.
```- name: Create SyncIQ policy
  dellemc_powerscale_synciqpolicy:
    onefs_host: "{{ dr_onefs_host }}"
    port_no: "{{ dr_port_no }}"
    verify_ssl: "{{ verify_ssl }}"
    api_user: "{{ dr_api_user }}"
    api_password: "{{ dr_api_password }}"
    action: "copy"
    enabled: True
    policy_name: "{{ item.policy_name }}"
    description: "{{ item.description }}"
    run_job: "when-snapshot-taken"
    skip_when_source_unmodified: False
    snapshot_sync_pattern: "*"
    source_cluster:
      source_root_path: "{{ item.source_root_path }}"
      source_include_directories: "{{ item.source_include_directories_list }}"
    target_cluster:
      target_host: "{{ item.target_host }}"
      target_path: "{{ item.target_path }}"
    state: "present"
  with_items:
    "{{ sasktel.isilon.dr.synciqs }}"```

And here is the error message:
```failed: [localhost] (item={'policy_name': 'Ansible_policy', 'description': 'Policy created by Ansible', 'source_root_path': 
'/ifs/data/Mike', 'source_include_directories_list': ['/ifs/data/Mike/SOURCE/Folder1'], 'target_host': 'localhost', 'target_path': 
'/ifs/data/RESTORED', 'filename': 'file1.txt', 'date': '2022-01-01', 'source_snapshot': 'MikeTestRestore', 'retries': 200}) => 
{"ansible_loop_var": "item", "changed": false, "item": {"date": "2022-01-01", "description": "Policy created by Ansible", "filename": 
"file1.txt", "policy_name": "Ansible_policy", "retries": 200, "source_include_directories_list": ["/ifs/data/Mike/SOURCE/Folder1"], 
"source_root_path": "/ifs/data/Mike", "source_snapshot": "MikeTestRestore", "target_host": "localhost", "target_path": 
"/ifs/data/RESTORED"}, "msg": "Get details of SyncIQ policy Ansible_policy failed with error : Invalid value for 
`snapshot_sync_pattern`, length must be greater than or equal to `1`"}
```
No matter what I do, it gives me the same error even though you see that I have the **snapshot_sync_pattern** included.
Maybe there is something else I need to include?  I noticed that snapshot_sync_pattern is NOT in any of the examples.
Maybe you guys are running a different version?
If you want we can do a zoom session and I can show you.  I also have a Dell SR open.

**Comment by dcsk1234** [comment]
Hi Glenn,
Thanks for sharing create task playbook syntax. We have tried to create policy according to shared syntax but we are getting validation errors because of following reasons.
1. "skip_when_source_unmodified" is applicable only when run_job:'on-schedule'
2. target_certificate_id is missing under target_cluster

As per implementation logic, first step in create task is to fetch list of all existing policies to determine if the current policy exists or not.so when this retrieval is happening, there could be some policy which is having snapshot_sync_pattern with blank value and which is causing the task to fail.

To analyse and understand the issue further shall we have a Zoom session?

Regards,
Krishna

**Comment by davisglenn** [comment]
Krishna,
Yes, that would be great to have a Zoom call because as I showed at the beginning of the issue, I can't even do a GET on an existing Policy.
That also fails.  The error doesn't say that the Policy doesn't exist. Instead I only get snapshot_sync_pattern errors. No matter what I do.
Let me know the latest you can meet and I will show you what I've seen.
thanks.
glenn.

**Comment by dcsk1234** [comment]
Hi Glenn,
As discussed, Zoom meeting is scheduled and invite is sent. Pls do let us know if incase meeting schedule requires to be changed

Regards,
Krishna

**Comment by anupamaloke** [comment]
@davisglenn This issue has been fixed in v1.5.0. Please re-open or create a new issue if you are still running into this error with the latest version.
